### PR TITLE
QCA upgrade for future botan2 compat (see trac #58567)

### PR DESCRIPTION
**Pull request by RJVB**
I've given the portfile an overhaul and much-needed cleanup, and introduced a subport for the (new?) PKCS11 plugin.

All subports have been tested on 10.9.5 against Qt 4.8.7 and 5.9.8 .

### CODE REVIEW

File: devel/qca/Portfile [hunk 1]
```diff
@@ -1,61 +1,49 @@
...
+                        (BlowFish, 3DES, AES).  Functionality is supplied via plugins.
+
+platforms               darwin
+
+homepage                https://userbase.kde.org/QCA
+master_sites            https://download.kde.org/stable/qca/${version}
+use_xz                  yes
+checksums               rmd160  dddc3cf240dc5424b9df13fc1bf41c8e04f3b814 \
+                        sha256  d716d2d8e3ed8d95bbdb061f03081d7d032206f746a30a4d29d72196f50e7b02
+#                         size    691676
```
Comment (yan12125) [inline devel/qca/Portfile]: Is there a reason that this line is commented out?

File: devel/qca/Portfile [hunk 2]
```diff
@@ -1,61 +1,49 @@
...
-
-# remove unnecessary Carbon #include, which fails on 10.8
-patchfiles-append   patch-src_qac_systemstore_mac.cpp.diff
-
-patchfiles-append   patch-qca-ossl.diff
+# best included after the required Qt PG
+PortGroup               cmake 1.1
+
+categories              devel crypto security
+maintainers             gmail.com:rjvbertin openmaintainer
```
Comment (yan12125) [inline devel/qca/Portfile]: Could you include your GitHub handle @RJVB here?

**Comment by RJVB** [comment]
On Sunday June 09 2019 07:57:54 Chih-Hsuan Yen wrote:

>Is there a reason that this line is commented out?
>Could you include your GitHub handle @RJVB here?

I keep my own ports tree free of such useless redundancies, and tend to forget to put them back in PRs

**Comment by yan12125** [comment]
As a record, I added a commit for fixing qca-qt5 on 10.8: https://github.com/macports/macports-ports/commit/f592ed6b2cf6616b0dfc4545090675eb9786bdbd. That's a small change so I skipped the usual pull request flow.

**Comment by RJVB** [comment]
>As a record, I added a commit for fixing qca-qt5 on 10.8: https://github.com/macports/macports-ports/commit/f592ed6b2cf6616b0dfc4545090675eb9786bdbd. That's a small change so I skipped the usual pull request flow.


That's something the Qt5 Portgroup should take care of, don't you think?

**Comment by yan12125** [comment]
Yeah, I plan to fix the qt5 port group after botan2 tasks.

**Comment by RJVB** [comment]
>Yeah, I plan to fix the qt5 port group after botan2 tasks.

That's a complex PG so probably best leave this to the Qt5 maintainer, or else use a PR for that change.

**Comment by RJVB** [comment]
>That's something the Qt5 Portgroup should take care of, don't you think?

Come to think of it, Qt 5.7 does not yet require C++11, and Qt 5.9 (which does) requires 10.10 which doesn't require the cxx11 PG. So with hindsight it is probably NOT necessary that the Qt5 PG includes the cxx11 PG.

Is there a lot of C++11 syntax in the qca-qt5 code?

**Comment by yan12125** [comment]
Qt 5.7 does require C++ 11. An example broken build can be found at https://build.macports.org/builders/ports-10.8_x86_64_legacy-builder/builds/96513/steps/install-port/logs/stdio. Qt Wiki says so, too [1].

I checked into more details but I got confused. As per the latest macports-base, C++ 11 is supported since Xcode 5.0 [2], and the 10.8 buildbot is installed with Xcode 5.1.1. Maybe Qt 5.7 requires more than C++ 11.

[1] https://wiki.qt.io/New_Features_in_Qt_5.7
[2] https://github.com/macports/macports-base/blob/master/src/port1.0/portconfigure.tcl#L657

**Comment by RJVB** [comment]
On Friday June 14 2019 09:23:17 Chih-Hsuan Yen wrote:
> Qt 5.7 does require C++ 11. 

For building Qt itself yes. Maybe qca depends on some private stuff which means C++11 support is required for it. Strange, I can imagine you got confused...

**Comment by yan12125** [comment]
I'll wait to see if there is another port broken with similar error messages on Qt 5.7 and OS X 10.8. Maybe you're right - it's not a general Qt issue.

**Pull request merged**